### PR TITLE
Add PyLint with docstring linter

### PR DESCRIPTION
**Pull request by adeadman**

### CODE REVIEW

File: pyticketswitch/availability.py [hunk 1]
```diff
@@ -203,9 +203,9 @@ def is_available(self, year, month=None, day=None):
         """Check availablity on a given year/month/day.
 
         Args:
-            year (int): the year to check. E.G.: 2005, 2016, 2222, etc
-            month (int, optional): the month to check. 1 == jan, 12 == dec, etc
-            day (int, optional): the day of the month, zero indexed.
+            year (int): the year to check. For example, 2005, 2016, 2222, etc
+            month (int): (optional) the month to check. 1 == jan, 12 == dec, etc
+            day (int): (optional) the day of the month, zero indexed.
```
Comment (nicwest) [inline pyticketswitch/availability.py]: I thought we were ditching these?

File: Makefile [hunk 2]
```diff
@@ -16,6 +16,7 @@ clean:
 
 test:
 	flake8 pyticketswitch
```
Comment (nicwest) [inline Makefile]: is there any overlap between pylint and flake8? do we need to run both?

File: pyticketswitch/availability.py (hunk 1 above)
Comment (adeadman) [inline pyticketswitch/availability.py]: Can do, although there's some code in this method that handles the case when no month or day is supplied. I think this was changed before that chat so it slipped through, but I don't mind either way.

File: Makefile (hunk 2 above)
Comment (adeadman) [inline Makefile]: There is overlap, but I have configured pylint to only check for docstring problems. If the default settings are used it complains about way more stuff (a lot of which we don't care about, such as module-level docstrings, code formatting, etc) - more fine tuning could be useful and at some point we may be able to ditch flake8 if pylint can do everything we use it for.

**Pull request merged**